### PR TITLE
Allow Id Console to rename job titles (#5815)

### PR DESCRIPTION
**Pull request by vlandemart**
### Purpose
CL: [Improvement] Allows ID Console to rename occupation to allow custom job titles
Fixes last point in #5815 

### Notes
This change was tested on local server with two clients

### Changelog:
* ID card now displays Occupation.DisplayName instead of ENUM converted to string (Goodbye, CAPTAIN, welcome Captain!)
* Added new TextField to GUI ID Console to allow renaming JobTitle
* Changes all ID.JobType.ToString() usages to ID.GetJobTitle()

### CODE REVIEW

File: UnityProject/Assets/Scripts/Items/IDCard.cs [hunk 1]
```diff
@@ -162,10 +175,16 @@ private void RenameIDObject()
 			newName += $"{RegisteredName}'s ";
 		}
 		newName += "ID Card";
-		if (JobType != JobType.NULL)
+
+		if (!jobTitle.IsNullOrEmpty())
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Items/IDCard.cs]:
```suggestion
		if (jobTitle.IsNullOrEmpty() == false)
```

File: UnityProject/Assets/Scripts/Items/IDCard.cs [hunk 2]
```diff
@@ -197,6 +216,11 @@ public bool HasAccess(Access access)
 		return accessSyncList.Contains((int) access);
 	}
 
+	public string GetJobTitle()
+	{
+		return jobTitle.IsNullOrEmpty() ? jobType.ToString() : jobTitle;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Items/IDCard.cs]: maybe it could get JobTitle from the job SO's by comparing the JobType

Comment (vlandemart) [reply UnityProject/Assets/Scripts/Items/IDCard.cs]: You mean AllowedOccupationsSingleton.asset, or is there another SO for jobs?

Comment (corp-0) [reply UnityProject/Assets/Scripts/Items/IDCard.cs]: AllowedOccupationsSingleton should have a list of Occupations which is the thing I'm talking about

**Review by corp-0 — APPROVED** [review]

**Review by ThatDan123 — APPROVED** [review]

**Pull request merged**